blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,361 @@
+cryptographic algorithm and sizes 2048, 3072, and 4096. Keys generated with the
+ECDSA cryptographic algorithm are not correctly supported.
+- The firmware does not verify the expiration date of certificates during the
+verification of launched files. It means that enrolled certificates may expire
+and will not affect the ability to boot files verified by them.
+- The firmware only allows the resetting of enrolled certificates when they
+have been added from the UEFI BIOS Menu. If they are added from the operating
+system, the certificates are marked as External, and only a full firmware
+reset (for example, by removing the CMOS battery) allows their removal.
+- Certificates enrolled through the Automatic Certificate Provisioning method
+are correctly used to verify launched files.
+- The [sbctl](https://github.com/Foxboron/sbctl) tool can be used to manage UEFI
+Secure Boot certificates.
+- Automatic tests of the sbctl tool and the Automatic Certificate Provisioning
+methods have been omitted due to their logic, assuming that firmware can remove
+certificates enrolled this way from the UEFI BIOS Menu.
+
+The conclusions allowed us to proceed with UEFI Secure Boot integration in the
+Yocto layer.
+
+## Integrating Secure Boot into a Yocto layer
+
+Integrating UEFI Secure Boot into an existing Yocto layer is possible by using
+the [meta-secure-core](https://github.com/Wind-River/meta-secure-core) layer in
+your build. Its sublayer – `meta-efi-secure-boot` introduces mechanisms that
+allow verifying various files used in the boot process. It offers two secure
+boot technologies:
+
+- UEFI Secure Boot, which verifies images loaded by UEFI firmware against
+certificates
+- MOK (Machine Owner Key) Secure Boot, which extends UEFI SB by
+introducing user-added Machine Owner Keys
+
+### Custom certificates
+
+UEFI Secure Boot allows the user to enroll custom certificates which will be
+used to verify images. The certificates can be loaded automatically thanks to
+the
+[Automatic Certificate Provisioning](https://github.com/Wind-River/meta-secure-core/tree/master/meta-efi-secure-boot#automatic-certificate-provision)
+procedure provided by `meta-secure-core`. The procedure uses `LockDown.efi`,
+which can only be executed when UEFI Secure Boot is disabled. `LockDown.efi`
+provisions the new certificates, which are built into it during build.
+
+UEFI Secure Boot uses the DB certificate to verify GRUB. Aside from that,
+`grubenv` and `grub.cfg` are signed during build using a boot key, which can
+be defined in the layer. GRUB then tries to load the kernel, which is also
+checked against DB. If any of the checks fail, the respective file cannot be
+booted. This mechanism increases the system's security by making sure that the
+firmware does not boot untrusted files.
+
+### Implementation
+
+To integrate the mechanisms available in `meta-secure-core` you need to
+integrate that layer into your build. We use
```

Comment:
@PLangowski Sometimes I get lost whether we are writing about something we did (we use) vs writing for the reader to do something (you need to). It would be useful for the text to be consistent